src/support/modify.c

```diff
@@ -425,6 +409,32 @@ __wt_modify_apply_item(WT_SESSION_IMPL *session, WT_ITEM *value, const void *mod
     return (0);
 }
 
+/*
+ * __wt_modify_apply_buf --
+ *     Apply a single set of WT_MODIFY changes to a buffer.
+ */
+int
+__wt_modify_apply_buf(WT_SESSION_IMPL *session, WT_ITEM *buf, const void *modify)
+{
+    return (__modify_apply_item(session, buf, modify, S2BT(session)->value_format[0] == 'S'));
+}
+
+/*
+ * __wt_modify_apply --
+ *     Apply a single set of WT_MODIFY changes to a cursor buffer.
+ */
+int
+__wt_modify_apply(WT_CURSOR *cursor, const void *modify)
+{
+    WT_SESSION_IMPL *session;
+    bool sformat;
+
+    session = (WT_SESSION_IMPL *)cursor->session;
+    sformat = cursor->value_format[0] == 'S';
```

Comment:
I'd like an assert that the value format coming in here is only ever a single character. i.e WT_ASSERT(session, value_format[1] == `\0`);

Along with a comment describing why we care (which I outlined yesterday).

Reply:
I think we don't have that assert for a reason:

```
(gdb) f 1
#1  0x00007fbef09a820e in __wt_modify_apply_api (cursor=0x5562538f6980, entries=0x5562538fa998, nentries=1) at ../src/support/modify.c:428
428	    WT_ERR(
(gdb) f 0
#0  0x00007fbef09a7e56 in __wt_modify_apply_item (session=0x5562537f72e0, value=0x5562538f6aa8, modify=0x5562538f6b80) at ../src/support/modify.c:369
369	    WT_ASSERT(session, S2BT(session)->value_format[1] == '\0');
(gdb) p ((WT_BTREE *)(session)->dhandle->handle)
Cannot access memory at address 0x98
(gdb) p session->dhandle
$1 = (WT_DATA_HANDLE *) 0x0
```
Sometimes we call modify without the dhandle open.

Reply:
I need to revert back to my previous solution. We can't fold the check into __wt_modify_apply_api